### PR TITLE
Split Qt 4 into two formulae and rename it

### LINKED ISSUE #5: Rename qt to qt4
I recommend you rename the Qt formula to Qt4 and change `depends_on` accordingly. This is because now that Qt 4 has been removed from Homebrew in https://github.com/Homebrew/homebrew-core/pull/8306/ at some point in the future we'll be renaming the Qt5 formula to just Qt. I'll wait for your 👍 that you're ready before we do that to avoid screwing your tap.

### LINKED ISSUE #12: qt4: set default to -with-qt3support
Is it too much trouble to enable qt3support by default in the qt4 formula and bottles?

I know qt3support it was not enabled by default when the formula was in homebrew-core. 
It was however shipped in the official Qt binaries, which no longer work with macOS newer than 10.10.
Right now out Travis CI is at osx_image: xcode6.4 and pulling the official Qt .dmg to keep things running. 

I guess our project is the only one still stuck with qt3support... In this case I will have to fork your formula and make bottles myself until we get rid of qt3support.

### LINKED ISSUE #14: qt: split into multiple formulae
#12 gave me an idea: what if we split `qt` into multiple formulae? We could have `qtcore`, `qtwebkit`, `qt3support`, and so on. That would likely solve our bottle building issue, since we could make a separate Travis CI job for each formula and avoid the timeout.

Ubuntu appears to do this, with packages like [`libqt4-qt3support`](http://packages.ubuntu.com/precise/libqt4-qt3support) and [`libqtgui4`](http://packages.ubuntu.com/precise/libqtgui4). It's worth investigating further to see if this is viable on macOS.

### PR DESCRIPTION
**Pull request by cartr**
This Pull Request:

 - Closes #5 by renaming `qt` to `qt@4`.
 - Closes #14 by disabling WebKit support in `qt@4` and creating a `qt-webkit@2.3` formula with a newer version of WebKit. The WebKit source I'm using is still pretty old, but it's significantly more recent than the one Qt 4.8 ships with. Now both `qt@4` and `qt-webkit@2.3` should build quickly enough that we can use Travis to automatically make bottles.
 - Closes #12 by unconditionally enabling Qt3Support in `qt@4`.

What's left to do before this can be merged:

- [x] Qmake projects that use WebKit fail to build unless you manually edit the Makefile to add `qt-webkit@2.3`'s cellar to the include and framework search paths. I suspect this can be solved by patching lines 4769-4782 of the configure scripts to tell the built qmake to use /usr/local instead of `qt@4`'s Cellar prefix.
- [ ] Building binary bottles using Travis
- [ ] Ensuring `pyqt` and `pyside@1.2` still build correctly and can see WebKit when it's installed.
- [x] Ensuring users who currently have `cartr/qt4/qt` installed and run `brew upgrade` correctly receive `qt@4` and `qt-webkit@2.3`.
- [x] Testing to make sure third-party Qt 4 projects still build and work correctly
  - [x] FreeCAD Homebrew formula
  - [x] Non-Homebrew program that uses `qmake`
  - [x] Non-Homebrew program that uses `cmake`
- [x] Add `revision` tag to all formulae that hard-code `qt` as the formula name somewhere. (is this all of them?)

### CODE REVIEW

File: qt@4.rb [hunk 1]
```diff
@@ -134,11 +142,10 @@ def caveats; <<-EOS.undent
     We agreed to the Qt opensource license for you.
     If this is unacceptable you should uninstall.
 
-    Qt Designer no longer picks up changes to the QT_PLUGIN_PATH environment
-    variable as it was tweaked to search for plug-ins provided by formulae in
-      #{HOMEBREW_PREFIX}/lib/qt4/plugins
-
     Phonon is not supported on macOS Sierra or with Xcode 8.
+    
+    Webkit is no longer included -- please use `brew install qt-webkit@2.3` to
```
Comment (MikeMcQuaid) [inline qt@4.rb]: People will blindly run this and ignore the messaging in that formula that tells them not to.

File: qt-webkit@2.3.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+class QtWebkitAT23 < Formula
+  desc "Qt port of WebKit (insecure, you shouldn't use it)"
```
Comment (MikeMcQuaid) [inline qt-webkit@2.3.rb]: Drive-by comment: people will use it. If you don't think they should: don't package it or you're at least partially responsible for any security issues they find themselves affected by.

Comment (cartr) [reply qt-webkit@2.3.rb]: I've updated the warnings to make them more clear. In my opinion, the majority of `qt-webkit@2.3`'s uses (help files, content provided by the user in e.g. a trusted KML document) are perfectly fine -- it's just untrusted third-party Web content that should be avoided.

**Comment by cartr** [comment]
I'm working on getting PyQt to see WebKit when it's installed, and I've encountered really strange behavior. PyQt's configure script runs this command as part of a Makefile to test if QtWebKit is installed:

    g++ -c -pipe -O2 -arch x86_64 -Xarch_x86_64 -mmacosx-version-min=10.5 -Wall -W -DQT_NO_DEBUG -DQT_WEBKIT_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/local/Cellar/qt@4/4.8.7/mkspecs/macx-g++ -I. -I/usr/local/Cellar/qt@4/4.8.7/lib/QtCore.framework/Versions/4/Headers -I/usr/local/Cellar/qt@4/4.8.7/lib/QtCore.framework/Versions/4/Headers -I/usr/local/Cellar/qt@4/4.8.7/lib/QtNetwork.framework/Versions/4/Headers -I/usr/local/Cellar/qt@4/4.8.7/lib/QtNetwork.framework/Versions/4/Headers -I/usr/local/Cellar/qt@4/4.8.7/lib/QtGui.framework/Versions/4/Headers -I/usr/local/Cellar/qt@4/4.8.7/lib/QtGui.framework/Versions/4/Headers -I/usr/local/Cellar/qt-webkit@2.3/2.3.4/include/QtWebKit -I/usr/local/include -I. -F/usr/local/lib -o cfgtest_QtWebKit.o cfgtest_QtWebKit.cpp

Notice that `-I/usr/local/Cellar/qt-webkit@2.3/2.3.4/include/QtWebKit` is included, so the compiler should be able to find QtWebKit's header files. When I run this command manually in a `brew install --interactive` shell, it succeeds without issue. However, when the Makefile runs it, it fails:

```
cfgtest_QtWebKit.cpp:1:10: fatal error: 'qwebpage.h' file not found
#include <qwebpage.h>
         ^
1 error generated.
make: *** [cfgtest_QtWebKit.o] Error 1
```

[Running gcc with arguments to print out the include paths](https://gist.github.com/cartr/b4aa526ba99ed0a9841dbdd829785c93) reveals that gcc seems to just be ignoring some of its `-I` arguments when run via `make`. What?

**Comment by cartr** [comment]
Homebrew core developers [have indicated](https://github.com/Homebrew/homebrew-core/pull/8334#issuecomment-292099625) that they are unwilling to give me additional time to complete these changes and will likely be renaming their formulae on top of mine later tonight. To prevent breakage, I'm merging this now, even though the bottle builds have not yet finished.

Installing or upgrading the formulae in this tap will probably take a long time (~= 2 hours of compiles) until sometime tomorrow when I am able to finish building the binaries. I apologize for any inconvenience this causes.

File: formula_renames.json [hunk 3]
```diff
@@ -2,5 +2,7 @@
   "shiboken"     : "shiboken@1.2",
   "pyside"       : "pyside@1.2",
   "pyside-tools" : "pyside-tools@1.2",
-  "coin"         : "coin@3.1.3"
+  "coin"         : "coin@3.1.3",
+  "qt"           : "qt-legacy-formula",
```
Comment (MikeMcQuaid) [inline formula_renames.json:6]: Should be `qt@4` here?

Comment (cartr) [reply formula_renames.json:6]: No, as that would break everyone who was using QtWebkit. `qt-legacy-formula` installs `qt@4` and `qt-webkit@2.3`, and warns people that they likely need to re-run `cmake`/`qmake` on all the external software they use.

Comment (MikeMcQuaid) [reply formula_renames.json:6]: Ok, cool, I get it now. Not seen that pattern used before.
